src/olympia/scanners/views.py

```diff
@@ -16,9 +19,31 @@ class ScannerResultViewSet(ListAPIView):
         GroupPermission(amo.permissions.ADMIN_SCANNERS_RESULTS_VIEW)
     ]
 
-    queryset = ScannerResult.objects.all()
     serializer_class = ScannerResultSerializer
 
+    def get_queryset(self):
```

Comment:
So the issue is, this is going to potentially give you duplicate results, because we don't prevent reviewers from confirming an approval that has already been confirmed. Is that a problem ? I am not completely sure I get why we have an API that lists *all* results in the first place - eventually there is going to be a ton of them to paginate through ?

Reply:
ah, I see so we would need some sort of `DISTINCT` then?

> I am not completely sure I get why we have an API that lists all results in the first place - eventually there is going to be a ton of them to paginate through ?

Mainly to provide fresh data to the ML folks without having to manually do it. In the future, we could have a filter to only return results with `id` greater than a value or something. We could also have a hard time constraint to not return too many results in the future.

Reply:
You can add a `distinct()` yeah. Performance is probably going to be terrible, but we can start with that.

Reply:
mmm so the duplicate results would happen because of the `VersionLog` entries, right? I tried to write a test case with multiple entries for the same user/version but it does not return duplicate results 😕

Reply:
I think we cannot have duplicate results because the main model is `ScannerResult`, no?

Reply:
Actually, if you add another `VersionLog` and then inspect `get_queryset()`, `good_results` will contain a duplicate if evaluated, but the `union()` will make it disappear.
```diff
diff --git a/src/olympia/scanners/tests/test_views.py b/src/olympia/scanners/tests/test_views.py
 index 2010f13ddc..944d9149c5 100644
 --- a/src/olympia/scanners/tests/test_views.py
 +++ b/src/olympia/scanners/tests/test_views.py
 @@ -78,6 +78,14 @@ def test_get(self):
              ),
              version=good_version_2,
          )
 +        VersionLog.objects.create(
 +            activity_log=ActivityLog.create(
 +                action=amo.LOG.CONFIRM_AUTO_APPROVED,
 +                version=good_version_2,
 +                user=user_factory(),
 +            ),
 +            version=good_version_2,
 +        )
          # result NOT labelled as "good" because action is not correct.
          version_3 = version_factory(addon=addon_factory())
          ScannerResult.objects.create(scanner=YARA, version=version_3)
```

```
ipdb> good_results                                                                                                                                                                   
<BaseQuerySet [<ScannerResult: ScannerResult object (518)>, <ScannerResult: ScannerResult object (519)>, <ScannerResult: ScannerResult object (519)>]>
ipdb> bad_results                                                                                                                                                                    
<BaseQuerySet [<ScannerResult: ScannerResult object (515)>]>
ipdb> good_results.union(bad_results)                                                                                                                                                
<BaseQuerySet [<ScannerResult: ScannerResult object (518)>, <ScannerResult: ScannerResult object (519)>, <ScannerResult: ScannerResult object (515)>]>
ipdb>
```

So I think that yeah, we don't need the distinct, but adding the test would be nice to make sure it never regress if/when we changed the implementation.

Reply:
ah, thanks for helping me on this! I was a bit confused..

Reply:
I updated one of the test cases.